### ISSUE TITLE
DOC: groupby by indexer to 'resample' data

### ISSUE THREAD

**Issue by tomchor** [pydata/pandas#13271]
Related to [this question](http://stackoverflow.com/questions/37396264/pandas-equivalent-of-resample-for-integer-index).

I would like to propose either to expand `.resample` to work on indexes that are integers or to create another method that does the same thing (maybe more general `.upsample` and `.downsample` methods).

If it was implemented as an expansion of resample I imagine something like:

    df = pd.DataFrame(np.random.randn(10,2))
    df.resample(5, np.std)

Which would produce:
```python
          0         1
0  1.184582  0.492113
5  0.533134  0.982562
```
which is a pretty forward downsample case. For upsampling maybe some discussion is needed on how to return the indexes.

This should be simple to program and I think it would be a very useful shortcut.

**Comment by jreback** [comment]
You mean like this?

```
In [6]: df.groupby(df.index / 5).std()
Out[6]: 
          0         1
0  0.324344  0.824444
1  1.088061  0.854456
```

**Comment by jreback** [comment]
So'll rerpupose this to be a doc issue, if you'd like to add an example to: http://pandas.pydata.org/pandas-docs/stable/groupby.html#examples

**Comment by tomchor** [comment]
Yes, that is exactly what I meant! Nobody in SO that saw the question (including me) could come up with that.

I'm new to github, but I'll look up how to contribute and I'll probably write an example of that.

Thanks a lot.

**Comment by jreback** [comment]
contributing docs are [here](http://pandas.pydata.org/pandas-docs/stable/contributing.html)

thanks!

**Comment by pfrcks** [comment]
@jreback Hi, I was working on this and have implemented the requirement. However while doing so
I utilized 
`python make.py clean`
`python make.py --no-api`

Followed by

`python make.py --single groupby`
As the example was added to groupby.
The contribution guide says "you must make sure not to commit the file deletions to your Git repository!"

Currently upon executing `git status`, it shows all of the rst files it deleted. I am unsure what to do now. Should I simply remove those deleted rst files being tracked or something else?

Thanks.
